clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheStateServiceImpl.java

```diff
@@ -242,6 +250,139 @@ private ResourcePageSource createPageSource(String poolName, ServerSideConfigura
+      @Override
+      public Number call() throws Exception {
+        return store.occupiedMemory();
+      }
+    });
+
+    registerStatistic(store, storeName, "dataOccupiedMemory", STATISTICS_STORE_TAG, PROPERTY_STORE_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return store.dataOccupiedMemory();
+      }
+    });
+
+    registerStatistic(store, storeName, "entries", STATISTICS_STORE_TAG, PROPERTY_STORE_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return store.longSize();
+      }
+    });
+
+    registerStatistic(store, storeName, "usedSlotCount", STATISTICS_STORE_TAG, PROPERTY_STORE_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return store.usedSlotCount();
+      }
+    });
+
+    registerStatistic(store, storeName, "dataVitalMemory", STATISTICS_STORE_TAG, PROPERTY_STORE_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return store.dataVitalMemory();
+      }
+    });
+
+    registerStatistic(store, storeName, "vitalMemory", STATISTICS_STORE_TAG, PROPERTY_STORE_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return store.vitalMemory();
+      }
+    });
+
+    registerStatistic(store, storeName, "reprobeLength", STATISTICS_STORE_TAG, PROPERTY_STORE_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return store.reprobeLength();
+      }
+    });
+
+    registerStatistic(store, storeName, "removedSlotCount", STATISTICS_STORE_TAG, PROPERTY_STORE_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return store.removedSlotCount();
+      }
+    });
+
+    registerStatistic(store, storeName, "dataSize", STATISTICS_STORE_TAG, PROPERTY_STORE_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return store.dataSize();
+      }
+    });
+
+    registerStatistic(store, storeName, "tableCapacity", STATISTICS_STORE_TAG, PROPERTY_STORE_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return store.tableCapacity();
+      }
+    });
+
+  }
+
+  private void registerPoolStatistics(String poolName, ResourcePageSource pageSource) {
+    registerStatistic(pageSource, poolName, "allocatedMemory", STATISTICS_POOL_TAG, PROPERTY_POOL_KEY, new Callable<Number>() {
+      @Override
+      public Number call() throws Exception {
+        return pageSource.delegatePageSource.getAllocatedSizeUnSync();
+      }
+    });
+  }
+
+  private void unRegisterStoreStatistics(String storeName) throws InvalidStoreException {
+
+    ServerStoreImpl store = stores.get(storeName);
+    if (store == null) {
+      throw new InvalidStoreException("Clustered tier '" + storeName + "' does not exist");
```

Comment:
in this case, i guess it doesn't really matter if we throw.  We can just let the code continue.  If we don't have access to the store instance then there is no way we can remove it from the shadow tree.  Since the store instance has been removed from the stores map it should get removed from the shadow tree when the store instance gets garbage collected.  i will remove this throw statement.